### PR TITLE
Allows heads of staff to just connect to other holopads, rather than call.

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
Allows heads of staff (or, more specifically, anyone that has the same access red-alert requires) to force a connection through holopads. This forces the remote holopad to connect instantly, instead of waiting for the other side to pick up.

Certain "secure" holopads will not allow autoconnection, and will fall back on traditional calling. These holopads have replaced the standard ones in:
- The AI sat (but not along the outer catwalk) and upload.
- The Captain's office and quarters.
- The HoS' office, the armory and adjacent rooms, and the holopad inside the large security meeting room.

Secure holopads can be constructed by using a circuitboard that has been set to secure mode. This mode can be toggled using a multitool, in the same way that a freezer/heater board can be toggled with a screwdriver.

A few holopad placement changes have been made as well:
- Moved the HoS holopad on Metastation one tile to the west so as to not be under a chair at round-start.
- Added a secure holopad to the Cap's office and quarters on Donutstation.
- Added a secure holopad to the AI's chamber entrance on Pubby.
- Moved the AI's near-core holopad on Pubby one tile to the east so as to not be under a foam depenser.
- Added a secure holopad to the HoS' office on Pubby.

Also, a ping noise now plays when a holopad is connected to from another pad.

## Why It's Good For The Game
Allows heads of staff to use the holopads to keep tabs on the employees a bit more, and as a new way to remotely interact with them that doesn't involve radio use.

The original purpose of the secure holopads was to prevent heads of staff from just checking on the AI every round to catch a Malf. Since I went through the trouble of making them, though, I figured I'd add them to a few other places that thematically made sense.

## Changelog
:cl:
add: Heads of staff can now connect to holopads without the other side answering the call.
add: Secure holopads, which do not allow auto-connect, have replaced holopads in a very small number of places.
tweak: Placements of holopads in various areas on various stations have been slightly adjusted.
tweak: Holopads now ping when a connection from another pad is established.
/:cl:

~~I'm pretty sure I fucked up mapmerger somehow. But while I work on that, I figured I'd see if anyone had strong opinions about the rest of the PR.~~ is good now

### CODE REVIEW

**Comment by ATH1909** [comment]
God yes.

Could you give this ability to lawyers as well? I think that it'd suit them/their job's kit well.

**Comment by ATH1909** [comment]
Actually, could we just give this ability to everyone, not just heads of staff?

**Comment by ATH1909** [comment]
Oh, and while you're at it, could you make holopad disks record both sides of holopad conversations, instead of just one?

**Comment by zxaber** [comment]
The problem with everyone having it is that there's no real access restriction of which holopads you can connect to, aside from the "secure" ones that just disallow all auto-connections. While I think it's a fine perk for heads of staff to have, I don't know how I feel about every assistant being able to throw their hologram into random work areas.

(You can still call like normal, this just lets Heads of Staff bypass the call and auto-connect).

**Comment by cacogen** [comment]
yes good

**Comment by carlarctg** [comment]
the heck is this?? how do you connect to holothings

**Comment by zxaber** [comment]
Holopads now make a ping noise when called or connected to from another pad.

**Comment by FloranOtten** [comment]
Give Security this too. Be able to peek into a department before running off.

**Comment by anconfuzedrock** [comment]
could just make it so secure holopads have connection priveleges as well.

**Comment by ATH1909** [comment]
> The problem with everyone having it is that there's no real access restriction of which holopads you can connect to, aside from the "secure" ones that just disallow all auto-connections. While I think it's a fine perk for heads of staff to have, I don't know how I feel about every assistant being able to throw their hologram into random work areas.
> 
> (You can still call like normal, this just lets Heads of Staff bypass the call and auto-connect).

Okay, but could you please still give this to lawyers?

**Comment by zxaber** [comment]
>Give Security this too. Be able to peek into a department before running off.

Security has both the Warden and (usually) the AI to help them scope out a bust. (Truth be told, it would probably be less useful to broadcast your intentions by showing your hologram right before you break in).

>could just make it so secure holopads have connection priveleges as well.

The whole purpose of the secure holopads is that not even the captain can force them to connect. Otherwise, it would be a trivial thing to just check the AI's sat for hacked APCs about ten minutes into every round.

**Comment by ATH1909** [comment]
> Why do lawyers need the ability to force their hologram into random rooms?

Isn't being nosy part of their schtick? Plus, if you're only present and objecting to the war crimes that the sec team is committing in hologram form, sec can't arrest/beat into crit you (immediately).

I dunno, it just felt like this ability is in the same sort of category as the lawyer's PDA's ability to send a PDA message to everyone at once. I... don't recall why.

File: code/game/machinery/hologram.dm [hunk 1]
```diff
@@ -60,6 +61,18 @@ Possible to do for anyone motivated enough:
 	var/ringing = FALSE
 	var/offset = FALSE
 	var/on_network = TRUE
+	var/secure = FALSE //for pads in secure areas; do not allow forced connecting
+
+/obj/machinery/holopad/secure
+	name = "secure holopad"
+	desc = "It's a floor-mounted device for projecting holographic images. This one will refuse to auto-connect incomming calls."
```
Comment (kriskog) [inline code/game/machinery/hologram.dm]: incomming

File: code/datums/holocall.dm [hunk 2]
```diff
@@ -32,20 +32,29 @@
 	var/datum/action/innate/end_holocall/hangup	//hangup action
 
 	var/call_start_time
+	var/head_call = FALSE //calls from a head of staff autoconnect, if the recieving pad is not secure.
 
 //creates a holocall made by `caller` from `calling_pad` to `callees`
-/datum/holocall/New(mob/living/caller, obj/machinery/holopad/calling_pad, list/callees)
+/datum/holocall/New(mob/living/caller, obj/machinery/holopad/calling_pad, list/callees, var/elevated_access = FALSE)
```
Comment (kriskog) [inline code/datums/holocall.dm]: No need for the var/

File: code/datums/holocall.dm [hunk 3]
```diff
@@ -32,20 +32,29 @@
...
 	calling_holopad = calling_pad
+	head_call = elevated_access
 	dialed_holopads = list()
 
 	for(var/I in callees)
 		var/obj/machinery/holopad/H = I
 		if(!QDELETED(H) && H.is_operational())
 			dialed_holopads += H
-			H.say("Incoming call.")
+			if(head_call)
```
Comment (kriskog) [inline code/datums/holocall.dm]: head_call is only used here, just use elevated_access from the proc arguments directly

**Review by kriskog — CHANGES_REQUESTED** [review]

File: code/datums/holocall.dm (hunk 3 above)
Comment (zxaber) [reply code/datums/holocall.dm]: Line 385 of hologram.dm checks if head_call is set on the datum, so we'd need the head_call var either way.

Comment (kriskog) [reply code/datums/holocall.dm]: Ah, gotcha

File: code/game/machinery/hologram.dm (hunk 1 above)
Comment (zxaber) [reply code/game/machinery/hologram.dm]: Fixed, thanks

**Pull request merged**